index.js

```diff
@@ -397,6 +404,7 @@ const plugin = fp(async function (app, opts) {
         // this query errored
         const err = new MER_ERR_GQL_VALIDATION()
         err.errors = cachedError.validationErrors
+        span.end()
```

Comment:
Usually on erros error info is added to span as status code

```
const { SpanStatusCode } = require('@opentelemetry/api')

span.setStatus({
  code: SpanStatusCode.ERROR,
  message: error.message,
});
```

As well as `OK` status on success.
```
span.setStatus({
  code: SpanStatusCode.OK,
});
```

Also to avoid multiple `span.end()` I prefer following pattern:

```js
const result = tracer.startActiveSpan(
  `<span_name>`,
  { attributes: <spanAttributes> },
  async (span) => {
    try {
      // do something
      span.setStatus({ code: SpanStatusCode.OK });
      // return if needed
    } catch (error) {
      span.setStatus({
        code: SpanStatusCode.ERROR,
        message: error.message,
      });
      throw error;
    } finally {
      span.end();
    }
  },
);
return result;
```

But it might be little bit less performant. And this one is available in most recent opentelemetry API, not sure about version from package.json.